fearless/Common/Services/RewardCalculatorService/RewardCalculatorService.swift

```diff
@@ -0,0 +1,269 @@
+                                  executing closure: @escaping (RewardCalculatorEngineProtocol) -> Void) {
+        let request = PendingRequest(resultClosure: closure, queue: queue)
+
+        if let snapshot = snapshot {
+            deliver(snapshot: snapshot, to: request)
+        } else {
+            pendingRequests.append(request)
+        }
+    }
+
+    private func deliver(snapshot: BigUInt, to request: PendingRequest) {
+        let eraOperation = eraValidatorsService.fetchInfoOperation()
+
+        eraOperation.completionBlock = {
+            dispatchInQueueWhenPossible(request.queue) {
+                if let result = try? eraOperation.extractResultData() {
+                    if let chain = self.chain {
+                        let calculator = RewardCalculatorEngine(totalIssuance: snapshot,
+                                                                validators: result.validators,
+                                                                chain: chain)
+                        request.resultClosure(calculator)
+                    }
+                }
+            } }
+
+        operationManager.enqueue(operations: [eraOperation],
+                                 in: .transient)
+    }
+
+    private func notifyPendingClosures(with totalIssuance: BigUInt) {
+        logger.debug("Attempt fulfill pendings \(pendingRequests.count)")
+
+        guard !pendingRequests.isEmpty else {
+            return
+        }
+
+        let requests = pendingRequests
+        pendingRequests = []
+
+        requests.forEach { deliver(snapshot: totalIssuance, to: $0) }
+
+        logger.debug("Fulfilled pendings")
+    }
+
+    private func handleTotalIssuanceDecodingResult(chain: Chain, result: Result<String, Error>?) {
+        guard chain == self.chain else {
+            Logger.shared.warning("Total Issuance decoding triggered but chain changed. Cancelled.")
+            return
+        }
+
+        switch result {
+        case .success(let totalIssuance):
+            guard let value = BigUInt(totalIssuance) else {
+                logger.error("Can't decode total issuance \(totalIssuance)")
+                return
+            }
+
+            self.snapshot = value
+            notifyPendingClosures(with: value)
+        case .failure(let error):
+            logger.error("Did receive total issuance decoding error: \(error)")
+        case .none:
+            logger.warning("Error decoding operation canceled")
+        }
+    }
+
+    private func didUpdateTotalIssuanceItem(_ totalIssuanceItem: ChainStorageItem?) {
+        guard let chain = chain else {
+            logger.warning("Missing chain to proccess total issuance")
+            return
+        }
+
+        guard let totalIssuanceItem = totalIssuanceItem else {
+            return
+        }
+
+        let codingFactoryOperation = runtimeCodingService.fetchCoderFactoryOperation()
+        let decodingOperation = StorageDecodingOperation<String>(path: .totalIssuance,
```

Comment:
Now we can use StringScaleMapper (which is in develop already) to avoid manual mapping to the integer from string